src/components/Integrations/Shopify/Shopify.js

```diff
@@ -1,16 +1,33 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Helmet } from 'react-helmet';
+import { HardcodedShopifyClient } from '../../../services/shopify-client.doubles';
 
-const Shopify = () => (
-  <>
-    <Helmet>
-      <title>Doppler | Shopify</title>
-    </Helmet>
-    <section style={{ width: '100%', padding: '60px 30px' }}>
-      Panel de control | Integraciones y preferencias avanzadas
-      <h1>Panel de Control - Shopify</h1>
-    </section>
-  </>
-);
+const shopifyClient = new HardcodedShopifyClient();
+
+const Shopify = () => {
+  const [shopName, setShopName] = useState('');
+  useEffect(() => {
+    const getData = async () => {
+      const result = await shopifyClient.getShopifyData();
+      if (result.length) {
```

Comment:
What happens in the other case?

Reply:
You are clearly not connected, without shops that is.

Reply:
In that case, I think that you should make it explicit.

Maybe something like this:

```js
  const [shopName, setShopName] = useState(null);
  // shopName === null would mean "loading"
  useEffect(() => {
    const getData = async () => {
      const result = await shopifyClient.getShopifyData();
      if (result.length) {
        setShopName(result[0].shopName);
        // shopName === something would mean "there is a shop"
      } else {
        setShopName('');
        // shopName === '' would mean "there is not a shop"
    };
    getData();
  }, []);
```

I think that it could work, but maybe it would be better having more than a _state_ in order to make it more explicit. Or having something different, for example an object:


```js
  cont _thereIsNoShop = {};
  // . . .
  const [shop, setShop] = useState(null);
  // shop === null would mean "loading"
  useEffect(() => {
    const getData = async () => {
      const result = await shopifyClient.getShopifyData();
      if (result.length) {
        setShopName(result[0]);
        // shop would contains "the shop".
      } else {
        setShop(_thereIsNoShop);
        // shop === _thereIsNoShop would mean that there is no job
    };
    getData();
  }, []);
```